ideas/167-extended-automated-testing-for-continuous-delivery/README.md

```diff
@@ -0,0 +1,58 @@
+## Preamble
+
+    Idea: 167
+    Title: Functional automated testing as a part of Continuous Delivery process
+    Status: Draft
+    Created: 2018-04-12
+
+## Summary
+For automated testing to be a part of CD process we need to:
+- increase coverage
+- increase reliability
+- provide visible results on daily basis
+
+## Swarm Participants
+- Lead Contributor: @antdanchenko
+- Testing & Evaluation: @antdanchenko
+- Contributor: @yevh-berdnyk
+- Contributor: @lukaszfryc
+- PM:
+
+## Product Overview
+Extended set of automated tests which will run against each nightly build in order to provide results via TestRun in TestRail + short summary via Slack notification
+As result during nightly build testing a tester should execute only those tests which are not covered by automation in the same TestRun
+In case of failure in automated test a tester should add an appropriate status to the test (see `An option to review results should be added for each test via adding 4 statuses to TestRail` in `Product Description` section)
+Actual sanity test case which is running against each PR will keep running without changes
+
+
+### Product Description
+- Each automated test suite should be based on existing set of cases from TestRail for tracking percents of functional test coverage
+- Each automated test should be the reflection of a test case from TestRail
+- Results of each automated test from a build should be provided via TestRun in TestRail and linked to the appropriate case
+- An option to review results should be added for each test via adding 4 statuses to TestRail:
+  - feature changed (for changed product behavior)
+  - known bug (link to the issue)
+  - false failure (bug in test)
+  - infrastructure issue
+
+### Minimum Viable Product
+Goal Date: 2018-05-01
+
+- 80% of test cases for nightly build from TestRail are covered by automated tests
```

Comment:
Are you referring to covering 80% of **the current scenarios** from TestRail? If there are some scenarios in TestRail we are already missing we should add them first.

Reply:
I think we should freeze test cases in a week before goal date and refer to that state 

And in future with adding/removing functionality we will modify our tests in order to reflect future states. This could done in scope of working on 'feature changed' statuses in a nightly TestRun by nightly tester